### ISSUE TITLE
Generated secret key has zero iv for secret key data encryption

### ISSUE THREAD

**Issue by ni4** [riboseinc/rnp#230]
IV for secret key encryption is not initialized anywhere, so it is always zero. However it should be set to random value.

```
:secret key packet:
     version 4, algo 1, created 1497878188, expires 0
     pkey[0]: [2048 bits]
     pkey[1]: [17 bits]
     iter+salt S2K, algo: 3, SHA1 protection, hash: 8, salt: 0B50EF85FAEDE84F
     protect count: 65536 (96)
     protect IV:  00 00 00 00 00 00 00 00
     skey[2]: [v4 protected]
     keyid: 51DAFAFCF1BB172E
```

**Comment by ni4** [comment]
It initializes the salt, but not the initial vector.

**Comment by dewyatt** [comment]
@ni4 I edited your comment to use triple backticks, a bit easier to read.

So the IV is always zero since the structure is allocated with calloc. Does RFC4880 13.9 explain this? (I haven't really looked)

**Comment by ni4** [comment]
@dewyatt Thanks for editing.
 
13.9 seems to explain only the CFB mode for plaintext encryption, not the secret keys. As I remember from the previous experience it is one of the parts of the RFC which cannot be easily understood.
3.7.2.1 tells: 

> These are followed by an Initial Vector of the same length as the block size of the cipher for the decryption of the secret values, if they are encrypted, and then the secret-key values themselves.

Something similar is told in the 5.5.3:

>[Optional] If secret data is encrypted (string-to-key usage octet not zero), an Initial Vector (IV) of the same length as the cipher's block size.

Also, GnuPG uses non-zero IV for secret key encryption:
```
:secret key packet:
	version 4, algo 1, created 1497876701, expires 0
	pkey[0]: [2048 bits]
	pkey[1]: [17 bits]
	iter+salt S2K, algo: 7, SHA1 protection, hash: 2, salt: 8AC03D3A62895FF2
	protect count: 7864320 (206)
	protect IV:  90 b0 ca a1 a8 97 a5 ea 1f 85 41 44 b0 f1 e1 37
	skey[2]: [v4 protected]
	keyid: 4E0CD9911CF621AF
```

**Comment by dewyatt** [comment]
I did notice GnuPG uses random salt+IV, like you said, so we should probably be doing the same since we are aiming to be compatible.

The RFC is pretty vague and under-specified, so this is probably one of those areas where the real standard is the GnuPG implementation itself.

**Comment by ni4** [comment]
Fixed and merged in f1a60f4.
